pgtransaction/atomic.py

```diff
@@ -18,17 +18,18 @@ def __init__(
         retry,
     ):
         super().__init__(using, savepoint, durable)
-        self.connection = transaction.get_connection(self.using)
         self.isolation_level = isolation_level
         self.retry = retry
         self._used_as_context_manager = True
         self._validate()
 
     def _validate(self):
+        connection = transaction.get_connection(self.using)
```

Comment:
Great catch, of course when this is a decorator all of this happens at module load time.

Reply:
Yea it totally crossed my mind too until I saw a failure in a test. Thank goodness for the test suite!